src/magic_folder/test/test_web.py

```diff
@@ -986,6 +987,66 @@ def test_add_participant(self, folder_name, personal_dmd):
             )
         )
 
+    @given(
+        author_names(),
+        folder_names(),
+    )
+    def test_add_participant_invalid(self, author, folder_name):
+        """
+        Missing keys in 'participant' JSON produces error
+        """
+        local_path = FilePath(self.mktemp())
+        local_path.makedirs()
+        folder_config = magic_folder_config(
+            create_local_author(author),
+            local_path,
+        )
+
+        root = create_fake_tahoe_root()
+        # put our Collective DMD into the fake root
+        root._uri.data[folder_config["collective-dircap"]] = dumps([
```

Comment:
For future reference, there's `root.add_data` and `root.add_mutable_data` that don't use internals

Reply:
Hmm. I went and looked a the other tests, and it looks like they aren't using those either. I think there may be a mismatch between the API the fake provides, and what is useful for us. I suspect we should add some wrappers or extend it to have a more friendly API. I filed #436 to do that.